provider/consulcatalog/consul_catalog.go

```diff
@@ -138,8 +139,15 @@ func (p *Provider) watch(configurationChan chan<- types.ConfigMessage, stop chan
 	watchCh := make(chan map[string][]string)
 	errorCh := make(chan error)
```

Comment:
Why not add 1 buffer on this chan instead of the sync.Once?

Reply:
Adding buffers will reduce maintainability because a number of capacity is depended on the number of gorountines which spawned at other functions. By using `sync.Once`, it will no longer need to worry about thinking the number of the buffer and blocking of the channel.

Reply:
ok for me.

We need this bug fix in 1.5, can you do this PR on branch v1.5 ?

Reply:
Changed base branch to v1.5 and patched the fix to it.